aicsimageio/readers/czi_reader.py

```diff
@@ -175,14 +183,34 @@ def scenes(self) -> Tuple[str, ...]:
                 #
                 # If we didn't do this, the produced list would have 96 of the same
                 # string name making it impossible to switch scenes.
-                for meta_scene in meta_scenes:
+                duplicates = {}
+                for scene_idx, meta_scene in enumerate(meta_scenes):
                     shape = meta_scene.find("Shape")
                     if shape is not None:
                         shape_name = shape.get("Name")
                         scene_name = meta_scene.get("Name")
                         combined_scene_name = f"{scene_name}-{shape_name}"
                     else:
                         combined_scene_name = meta_scene.get("Name")
+                        # Some scene names can be unpopulated, for those we should fill
+                        # with filename-idx
+                        if combined_scene_name is None:
+                            fname_prefix = Path(self._path).stem
+                            combined_scene_name = f"{fname_prefix}-{scene_idx}"
+                        # Check for duplicated names
+                        if combined_scene_name not in duplicates:
+                            duplicates[combined_scene_name] = [scene_idx, 1]
+                        else:
+                            if duplicates[combined_scene_name][1] == 1:
+                                scene_names[
+                                    duplicates[combined_scene_name][0]
+                                ] += f"-{duplicates[combined_scene_name][1]}"
+
+                            duplicates[combined_scene_name][1] += 1
+
+                            combined_scene_name += (
+                                f"-{duplicates[combined_scene_name][1]}"
+                            )
```

Comment:
It might be easier to read to collect all duplicates and then have another loop that updates them after this one. Keep as is unless someone else says something.